### PR TITLE
Kotlin 1.0.0 beta support

### LINKED ISSUE #29: Kotlin 1.0 Beta Candidate support
Please update Fuel for Kotlin 1.0 Beta Candidate.

http://blog.jetbrains.com/kotlin/2015/10/kotlin-1-0-beta-candidate-is-out/

### PR DESCRIPTION
**Pull request by yoavst**
Applied code cleanup. fix #29 
Note: I didn't messed up with #27.

### CODE REVIEW

File: fuel/src/main/kotlin/fuel/util/Delegates.kt [hunk 1]
```diff
@@ -1,18 +1,19 @@
 package fuel.util
 
 import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KProperty
```
Comment (kittinunf) [inline fuel/src/main/kotlin/fuel/util/Delegates.kt]: I m kind of a bit surprise that we can do this without `kotlin.reflect` dependency.

**Comment by kittinunf** [comment]
I will test again on my machine. If everything is great, I will merge this in.

**Comment by yoavst** [comment]
`KProperty` is not refelection only, but a basic functionality is provided on stdlib.

**Comment by kittinunf** [comment]
Ok noted.

**Comment by yoavst** [comment]
They have deprecated `PropertyMetadata` for `KProperty`.

**Comment by kittinunf** [comment]
It works great on my machine. I'm merging this in. Thanks :+1:

**Pull request merged**